python/tutorials/01-vector-add.py

```diff
@@ -23,6 +23,10 @@
 import triton
 import triton.language as tl
 
+DEVICE = triton.runtime.driver.active.get_current_target().backend
+# when using hip devices, the device string in pytorch is "cuda"
+PYTORCH_DEVICE = "cuda" if DEVICE == "hip" else DEVICE
```

Comment:
Can you get rid of this by doing
```
DEVICE = torch.device(triton.runtime.driver.active.get_current_target().backend)
```
Does this work for both CUDA and HIP?

Reply:
I think with this you should be able to simplify quite a bit most of the code

Reply:
@lezcano done, could you restart CI?

Reply:
> DEVICE = torch.device(triton.runtime.driver.active.get_current_target().backend)

@lezcano this doesn't work for HIP. I guess because this way `HIP` device (`device(type='hip', index=0)`) is built anyway, and not `cuda` device. What doesn't work for PyTorch on AMD GPU.

I guess I should roll back to the previous version if you have no more ideas?

Reply:
see https://github.com/triton-lang/triton/pull/5286#discussion_r1873480666